src/object/cli_shard.c

```diff
@@ -1688,7 +1688,7 @@ dc_enumerate_cb(tse_task_t *task, void *arg)
 			D_DEBUG(DB_IO, "key size "DF_U64" too big.\n",
 				oeo->oeo_size);
 			enum_args->eaa_kds[0].kd_key_len = oeo->oeo_size;
-		} else if (rc == -DER_INPROGRESS || rc == -DER_TX_BUSY) {
+		} else if (rc == -DER_INPROGRESS || rc == -DER_TX_BUSY || rc == -DER_TX_RESTART) {
```

Comment:
"DER_TX_RESTART" is not a retryable error, instead, it is the up layer sponsor (application) to decide whether restart related transaction or not. That is different from object level RPC retry. Log it as "need retry" is some misguided. If we do not want to log it as "ERROR", we can use other 'TRACE" log message.

Reply:
i guess i can just add a separate debug message if that helps

Reply:
I mean that current log message is that "xxx may need retry". But for 'DTX_TX_RESTART', it is not retryable error. So if we add another log message, such as "xxx may need restart", that will be more clear.